API/API.mli

```diff
@@ -5666,7 +5666,7 @@ module Vernacinterp :
 sig
   type deprecation = bool
 
-  type vernac_command = Genarg.raw_generic_argument list -> unit -> unit
+  type vernac_command = Genarg.raw_generic_argument list -> Loc.t option -> unit
```

Comment:
@ejgallego I let @herbelin reply to you as he knows more about these things than me.

To me, it would help if I could look at a PR that would realize what you have in mind.